control/fem_api_extension/api/src/ExcaliburFemClient.cpp

```diff
@@ -872,102 +935,135 @@ void ExcaliburFemClient::stopAcquisition(void)
+      throw FemClientException((FemClientErrorCode) excaliburFemClientIllegalOperationMode,
+                                msg.str());
+    }
+      break;
 
-      // Wait until DMA engine has transferred out the number of frames read out by the
-      // ASIC control block. This loop will repeat until ten times the frame acquisition
-      // length, after which it will assume that the completion has timed out
-      bool acqCompletePending = true;
-      int numAcqCompleteLoops = 0;
-      int maxAcqCompleteLoops = 10;
+  } // End of switch
 
-      while (acqCompletePending && (numAcqCompleteLoops < maxAcqCompleteLoops))
-      {
-        FemAcquireStatus acqState = this->acquireStatus();
-        FEMLOG(mFemId, logINFO) << "Asynchronous stop of DMA acquisition loop: " << numAcqCompleteLoops
-            << " attempts, ACQ state: " << acqState.state << " sent BDs: " << acqState.totalSent;
+  // Wait until DMA engine has transferred out the number of frames read out by the
+  // ASIC control block. This loop will repeat until ten times the frame acquisition
+  // length, after which it will assume that the completion has timed out
+  bool acqCompletePending = true;
+  int numAcqCompleteLoops = 0;
+  int maxAcqCompleteLoops = 10;
 
-        if (acqState.totalSent >= framesRead * 2)
-        {
-          FEMLOG(mFemId, logDEBUG) << "DMA controller has transmitted " << framesRead << " frames OK";
-          acqCompletePending = false;
-        }
-        else
-        {
-          numAcqCompleteLoops++;
-          usleep(mAcquisitionTimeMs * 1000);
-        }
-      }
-      if (acqCompletePending)
-      {
-        FEMLOG(mFemId, logERROR) << "ERROR: DMA transfer of " << framesRead
-            << " failed to complete in expected time during async stop";
-      }
+  while (acqCompletePending && (numAcqCompleteLoops < maxAcqCompleteLoops))
+  {
+    FemAcquireStatus acqState = this->acquireStatus();
+    FEMLOG(mFemId, logDEBUG) << "Awaiting DMA completion: " << numAcqCompleteLoops
+        << " attempts, ACQ state: " << acqState.state << " sent BDs: " << acqState.totalSent;
 
+    if (acqState.totalSent >= framesRead * subFramesPerFrame)
+    {
+      FEMLOG(mFemId, logDEBUG) << "DMA controller has transmitted " << framesRead << " frames OK";
+      acqCompletePending = false;
     }
+    else
+    {
+      numAcqCompleteLoops++;
+      usleep(mAcquisitionTimeMs * 1000);
+    }
+  }
+  if (acqCompletePending)
+  {
+    FemAcquireStatus acqState = this->acquireStatus();
+    FEMLOG(mFemId, logWARNING) << "DMA transfer of " << framesRead
+        << " frames failed to complete in expected time during async stop";
+    FEMLOG(mFemId, logWARNING) << "Expected " << framesRead * subFramesPerFrame 
+      << " BDs TXed, got " << acqState.totalSent 
+      << " RX: top " << acqState.totalRecvTop << " bottom " << acqState.totalRecvBot;
   }
 
+//    }
+//  }
+
```

Comment:
What is this?

Reply:
Lonely stray brackets looking for friends... Now removed.